### PR TITLE
Minaw.wu

### PR DESCRIPTION
**Pull request by MW0808**
Get leaderboard use case with basic visuals and test

### CODE REVIEW

File: src/main/java/us/jonathans/Testing.java
Comment (ch-iv) [inline src/main/java/us/jonathans/Testing.java:1]: Uncommit this file

File: src/main/java/us/jonathans/Config.java [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 public class Config {
     public static final String APP_NAME = "Jacala (Mancala in Java)";
+    public static final String leaderboardFrame= "Leaderboard";
```
Comment (ch-iv) [inline src/main/java/us/jonathans/Config.java:5]: Use uppercase for constants and give it a more descriptive name like `LEADERBOARD_FRAME_NAME`

File: src/main/java/us/jonathans/Entities/Leaderboard.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                }
+            }
+        }
+
+        return sortedData;
+    }
+}
+
+
+
```
Comment (ch-iv) [inline src/main/java/us/jonathans/Entities/Leaderboard.java:53]: Remove whitespace

File: src/main/java/us/jonathans/Entities/Leaderboard.java
Comment (ch-iv) [inline src/main/java/us/jonathans/Entities/Leaderboard.java:1]: Adapt the leaderboard to the format described [here](https://github.com/ch-iv/mancala/pull/11/files#diff-09ac1f7ca3996bd78c1bc6ece6c43142946fea6d66c1eb7030b492655c4571b1)

File: src/main/java/us/jonathans/InterfaceAdapters/GetLeaderboardPresenter.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        return values;
+
+    }
+
+    public void prepareUpdateScreen(GetLeaderboardOutputData getLeaderboardOutputData) {
+        getLeaderboardViewModel.firePropertyChanged("leaderboard",
+                this.getLeaderboardData(getLeaderboardOutputData.getLeaderboard()));
+    }
+
```
Comment (ch-iv) [inline src/main/java/us/jonathans/InterfaceAdapters/GetLeaderboardPresenter.java:47]: Remove whitespace

File: src/main/java/us/jonathans/Test/TestGetLeaderboard.java
Comment (ch-iv) [inline src/main/java/us/jonathans/Test/TestGetLeaderboard.java:1]: All tests should go in `src/test/java/us/jonathans`. Turn it into a proper integration test or remove the file from the PR

File: src/main/java/us/jonathans/InterfaceAdapters/GetLeaderboardPresenter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public class GetLeaderboardPresenter implements GetLeaderboardOutputBoundary {
+    GetLeaderboardViewModel getLeaderboardViewModel;
+
+    public GetLeaderboardPresenter(GetLeaderboardViewModel getLeaderboardViewModel) {
+        this.getLeaderboardViewModel = getLeaderboardViewModel;
+    }
+
+    @Override
+    public String[][] getLeaderboardData(Leaderboard leaderboard) {
```
Comment (ch-iv) [inline src/main/java/us/jonathans/InterfaceAdapters/GetLeaderboardPresenter.java:19]: Write a unit test for this function

File: src/main/java/us/jonathans/Entities/Leaderboard.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        list.add(time);
+
+        data.put(username, list);
+    }
+
+    public Map<String, ArrayList<Integer>> getData(){
+        return sort("score");
+    }
+
+    public Map<String, ArrayList<Integer>> sort(String type) {
```
Comment (ch-iv) [inline src/main/java/us/jonathans/Entities/Leaderboard.java:24]: Write a unit test for this

File: src/main/java/us/jonathans/Entities/Leaderboard.java
Comment (ch-iv) [inline src/main/java/us/jonathans/Entities/Leaderboard.java:1]: This file should be moved to an `entities` package

**Review by ch-iv — CHANGES_REQUESTED** [review]
Great PR. Could you also adjust the naming of your packages to be lowercase separated by underscore (e.g `UseCase` -> `use_case`)

**Comment by ch-iv** [comment]
Also for future PRs, please give your branches more descriptive names that relate to the feature that's being implemented. An appropriate name for the branch for this PR would be `implement-leaderboard-usecase`

**Review by RichardrLin — APPROVED** [review]
Looks good. Great job

**Pull request merged**